polar-core/src/partial/partial.rs

```diff
@@ -1057,9 +1051,8 @@ mod test {
         let next = next_binding(&mut q)?;
         assert_partial_expressions!(next, "x" => "_this = y", "y" => "x = _this");
         let next = next_binding(&mut q)?;
-        assert_eq!(next[&sym!("x")], term!(1));
-        assert_eq!(next[&sym!("y")], term!(1));
-        //assert_query_done!(q);
+        assert_partial_expressions!(next, "x" => "1 = _this and 1 = y", "y" => "1 = x and 1 = _this");
```

Comment:
This case should be fixed by grounding bug.  We can't resolve it in the simplifier because it could require checking that a proposed value is compatible with other constraints.  I looked at the query output when the grounding happens:

```
1 = _x_25, BINDINGS: {_x_25 = _x_25 = x and y = y and _x_25 = y}
```

If we use the VM for grounding as proposed, binding `_x_25` to `1` and evaluating the constraints would bind `x` to 1 and `y` to 1.